### ISSUE TITLE
False positive for exception-message-attribute with custom exceptions

### ISSUE THREAD

**Issue by adiroiban** [PyCQA/pylint#1564]
### Steps to reproduce

I have this source

```
class CompatException(Exception):
    """
    An exception which should work on py2 and py3.
    """
    def __init__(self, message=''):
        self.message = message

    def __repr__(self):
        result = u'CompatException %s' % (self.message)
        return result.encode('utf-8')


try:
    raise CompatException('message here')
except CompatException as error:
    message = error.message
```

### Current behavior

```
pylint --py3k test.py 
************* Module test
W: 10,42: Exception.message removed in Python 3 (exception-message-attribute)
W: 17,14: Exception.message removed in Python 3 (exception-message-attribute)

------------------------------------------------------------
An exception occurred while rating: name 'no' is not defined
```

### Expected behavior

I was expecting that the exception-message-attribute is not raises as my exception is explicit about having the message argument.

### pylint --version output

```
pylint --version
pylint 1.7.1, 
astroid 1.5.3
Python 2.7.13 (4f815b5, May  2 2017, 11:54:59) 
[GCC 5.4.0 20160609]
```
